alibi_detect/cd/sklearn/classifier.py

```diff
@@ -0,0 +1,215 @@
+            are specified, `n_folds` is prioritized.
+        retrain_from_scratch
+            Whether the classifier should be retrained from scratch for each set of test data or whether
+            it should instead continue training from where it left off on the previous set.
+        seed
+            Optional random seed for fold selection.
+        use_calibration
+            Whether to use calibration. Whether to use calibration. Calibration can be used on top of any model.
+        calibration_kwargs
+            Optional additional kwargs for calibration.
+            See https://scikit-learn.org/stable/modules/generated/sklearn.calibration.CalibratedClassifierCV.html
+            for more details.
+        data_type
+            Optionally specify the data type (tabular, image or time-series). Added to metadata.
+        """
+        super().__init__(
+            x_ref=x_ref,
+            p_val=p_val,
+            preprocess_x_ref=preprocess_x_ref,
+            update_x_ref=update_x_ref,
+            preprocess_fn=preprocess_fn,
+            preds_type=preds_type,
+            binarize_preds=binarize_preds,
+            train_size=train_size,
+            n_folds=n_folds,
+            retrain_from_scratch=retrain_from_scratch,
+            seed=seed,
+            data_type=data_type
+        )
+
+        if preds_type not in ['probs', 'scores']:
+            raise ValueError("'preds_type' should be 'probs' or 'scores'")
+
+        self.meta.update({'backend': 'sklearn'})
+        self.original_model = model
+        self.use_calibration = use_calibration
+        self.calibration_kwargs = dict() if calibration_kwargs is None else calibration_kwargs
+        self.model = self._clone_model()
+
+    def _clone_model(self):
+        model = clone(self.original_model)
+
+        # equivalence between `retrain_from_scratch` and `warm_start`
+        if not self.retrain_from_scratch:
+            if hasattr(model, 'warm_start'):
+                model.warm_start = True
+                logger.warning('`retrain_from_scratch=False` sets automatically the parameter `warm_start=True` '
+                               'for the given classifier. Please consult the documentation to ensure that the '
+                               '`warm_start=True` is applicable in the current context (i.e., for tree-based '
+                               'models such as RandomForest, setting `warm_start=True` is not applicable since the '
+                               'fit function expects the same dataset and an update/increase in the number of '
+                               'estimators - previous fitted estimators will be kept frozen while the new ones '
+                               'will be fitted).')
+            else:
+                logger.warning('Current classifier does not support `warm_start`. The model will be retrained '
+                               'from scratch every iteration.')
```

Comment:
Slightly niche question, but is it the case that if a classifier doesn't support `warm_start` subsequent calls to `fit` are idempotent (i.e. can we expect any classifier state or weights to be always reset to the same initial values)? The answer is probably yes at it appears the API is unified in this way (e.g. [SGDClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html) supports `warm_start` which just takes the existing weights and keeps iterating if `True`).